pkgs/native_assets_cli/lib/src/code_assets/config.dart

```diff
@@ -15,15 +15,38 @@ import 'os.dart';
 /// code assets (only available if code assets are supported).
 extension CodeAssetBuildConfig on BuildConfig {
   /// Code asset specific configuration.
-  CodeConfig get codeConfig => CodeConfig(this);
+  CodeConfig get code => CodeConfig(this);
 }
 
-/// Extension to the [LinkConfig] providing access to configuration specific to
-/// code assets as well as code asset inputs to the linker (only available if
-/// code assets are supported).
-extension CodeAssetLinkConfig on LinkConfig {
-  /// Code asset specific configuration.
-  CodeConfig get codeConfig => CodeConfig(this);
+// Weird class that combines `CodeConfig` getters and a getter for code assets.
+class LinkCodeConfig implements CodeConfig {
```

Comment:
This is indeed a bit ugly, but I don't know a way around it...

Reply:
This doesn't seem very right to me. It mixes orthogonal concerns (code configuration and outputed assets)